src/saltext/azurerm/utils/azurerm.py

```diff
@@ -100,13 +100,20 @@ def _determine_auth(**kwargs):
 
     try:
         credentials = DefaultAzureCredential(authority=authority, **kwargs)
-    except ClientAuthenticationError:
-        raise SaltInvocationError(  # pylint: disable=raise-missing-from
-            "Unable to determine credentials. "
-            "A subscription_id with username and password, "
-            "or client_id, secret, and tenant or a profile with the "
-            "required parameters populated"
-        )
+    except:  # pylint: disable=W0702
+        if "client_id" and "tenant" and "secret" in kwargs:
+            credentials = get_identity_credentials(**kwargs)
+        else:
+            kwargs.pop("client_id")
+            try:
+                credentials = DefaultAzureCredential(authority=authority, **kwargs)
```

Comment:
**suggestion (non-blocking):** I'm not a huge fan of this modified workflow, not from an operational perspective, but a code structure perspective. It seems a lot like "let's try to auth. it failed, so let's try another way. that also failed, so let's try another way". It _feels_ like you could do some checking up front in order to "do the right thing" on the first try.